spec/requests/api/alerts_statues_spec.rb

```diff
@@ -0,0 +1,100 @@
+  end
+
+  it "edits new alert status" do
+    api_basic_authorize action_identifier(:alerts_statuses, :edit)
+    expect(alert_status.result).to be_falsey
+    run_post(alert_status_url, gen_request(:edit, "result" => true))
+    expect(response).to have_http_status(:ok)
+
+    expect(alert_status.reload.result).to eq(true)
+  end
+
+  it "get all alerts statuses by container provider" do
+    api_basic_authorize collection_action_identifier(:alerts_statuses, :providers_alerts)
+    container_provider.container_nodes << container_node
+    container_node.miq_alert_statuses << alert_status
+    alert_status.miq_alert = miq_alert
+    alert_status.save!
+    run_post(alerts_statuses_url, gen_request(:providers_alerts))
+    expect(response).to have_http_status(:ok)
+    first_provider = response.parsed_body["providers"].first
+    expect(response.parsed_body["providers"].count).to eq(1)
+    expect(first_provider["name"]).to eq(container_provider.name)
+    expect(first_provider["type"]).to eq(container_provider.class.to_s)
+    expect(first_provider["id"]).to eq(container_provider.id)
+    expect(first_provider["alerts"].count).to eq(1)
+    expect(first_provider["alerts"].first["description"]).to eq(miq_alert.description)
+  end
+
+  it "returns alert status state" do
+    api_basic_authorize collection_action_identifier(:alerts_statuses, :providers_alerts)
+    container_provider.container_nodes << container_node
+    container_node.miq_alert_statuses << alert_status
+    alert_status.miq_alert = miq_alert
+    alert_status.save!
+    miq_alert_status_state
+
+    run_post(alerts_statuses_url, gen_request(:providers_alerts))
+    first_provider = response.parsed_body["providers"].first
+
+    expect(response).to have_http_status(:ok)
+    expect(first_provider["alerts"].first["states"].first["comment"]).to eq(miq_alert_status_state.comment)
+    expect(first_provider["alerts"].first["states"].first["username"]).to eq(miq_alert_status_state.user.name)
+    expect(first_provider["alerts"].first["states"].first["assignee_username"]).to eq(miq_alert_status_state.assignee.name)
+  end
+
+  it "get all alerts statuses by infra provider" do
+    api_basic_authorize collection_action_identifier(:alerts_statuses, :providers_alerts)
+    infra_provider.vms << vm
+    vm.miq_alert_statuses << alert_status
+    alert_status.miq_alert = miq_alert
+    alert_status.save!
+    run_post(alerts_statuses_url, gen_request(:providers_alerts))
+    expect(response).to have_http_status(:ok)
+    first_provider = response.parsed_body["providers"].first
+    expect(response.parsed_body["providers"].count).to eq(1)
+    expect(first_provider["name"]).to eq(infra_provider.name)
+    expect(first_provider["type"]).to eq(infra_provider.class.to_s)
+    expect(first_provider["id"]).to eq(infra_provider.id)
+    expect(first_provider["alerts"].count).to eq(1)
+    expect(first_provider["alerts"].first["description"]).to eq(miq_alert.description)
```

Comment:
e.g.:

```ruby
expected = {
  "providers" => [
    a_hash_including(
      "name" => infra_provider.name,
      # "etc" => "etc"
    )
  ]
}
expect(response.parsed_body).to include(expected)
```

Also, to further my point above about use of `let` - you considered for instance the "name" attribute to be important enough to verify, but at no point in the test have we specified what it is. This can lead to false positives. It is better to verify against literals:

```ruby
it "does something" do
  infra_provider = FactoryGirl.create(:ems_infra, :name => "Fred")
  # etc....
  expected = {
    "providers" => [
      a_hash_including(
        "name" => "Fred",
        # etc...     
      )
  ]
end
```